modules/apps/frontend-data-set/frontend-data-set-taglib/src/main/java/com/liferay/frontend/data/set/taglib/internal/jaxrs/application/FDSApplication.java

```diff
@@ -170,6 +171,57 @@ public Set<Object> getSingletons() {
 		return singletons;
 	}
 
+	@Path("/fds/{fdsName}/custom-views/{fdsCustomViewId}/label")
+	@POST
+	public Response renameFDSCustomView(
+		@PathParam("fdsName") String fdsName,
+		@PathParam("fdsCustomViewId") String fdsCustomViewId,
+		@FormParam("customViewLabel") String fdsCustomViewLabel,
+		@Context HttpServletRequest httpServletRequest,
+		@Context ThemeDisplay themeDisplay) {
```

Comment:
Are you using `themeDisplay`?

Reply:
Yes. It's used internally in `PortletPreferencesFactoryUtil.getPortalPreferences(...)`, with [this impl](https://github.com/liferay/liferay-portal/blob/master/portal-impl/src/com/liferay/portlet/PortletPreferencesFactoryImpl.java#L317-L321). Without the `@Context` annotation, that line breaks with NPE.

I am not sure if there is a better way to setup context dependencies. I copied the `@Context` annotation from other endpoints.